src/Optical/Emitters/Directions.jl

```diff
@@ -31,6 +31,10 @@ Constant(::Type{T} = Float64) where {T<:Real}
 struct Constant{T} <: AbstractDirectionDistribution{T}
     direction::Vec3{T}
 
+    function Constant(dirx::T,diry::T,dirz::T) where{T<:Real}
+        return new{T}(Vec3(dirx,diry,dirz))
+    end
+
```

Comment:
@BrianGun this is copied this from the optimization branch so there shouldn't be any merge conflicts.